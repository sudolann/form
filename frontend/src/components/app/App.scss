@import '~antd/dist/antd.css';
$gray: #5A5A5A;

body {
  font-family: 'Montserrat', Arial, sans-serif;
  color: $gray;
  margin: 0 auto;
  background: #fff;
}

.container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.loading {
  width: 100%;
  height: 500px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.flex {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 80vh;
  width: 90vw;
}