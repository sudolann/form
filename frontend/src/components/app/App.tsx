import React, { FunctionComponent, ReactElement } from "react";
import { EventById } from "../eventById/EventById";
import { AddNewEventForm } from "../addNewEventForm/AddNewEventForm";
import { EventsList } from "../eventsList/EventsList";
import { Route, Switch } from "react-router-dom";
import { Navigation } from "../navigation/Navigation";
import "./App.scss";

export const App: FunctionComponent = (): ReactElement => {
  return (
    <div className="container">
    <Navigation />
      <Switch>
        <Route path="/" exact component={EventsList} />
        <Route path="/form" component={AddNewEventForm} />
        <Route path="/:eventId" component={EventById} />
      </Switch>
    </div>
  );
};
