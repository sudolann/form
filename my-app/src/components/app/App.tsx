import React, { FunctionComponent, ReactElement } from 'react';
import './App.scss';

import { AddNewEventForm } from '../addNewEventForm/addNewEventForm';
import { EventById } from '../eventById/EventById';
import { Route, Switch } from 'react-router-dom';

export const App: FunctionComponent = (): ReactElement => {
  return (
    <div className="container">
      <Switch>
        <Route to="/" exact component={AddNewEventForm} />
        <Route to="/:eventID" component={EventById} />
      </Switch>
    </div>
  );
};
