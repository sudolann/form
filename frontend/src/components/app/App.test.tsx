import React from 'react';
import { App } from './App';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';

describe('App', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(
      <Router>
        <App />
      </Router>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
