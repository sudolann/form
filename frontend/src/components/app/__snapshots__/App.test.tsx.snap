// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="nav"
    >
      <a
        aria-current="page"
        class="nav--link active"
        href="/"
      >
        Events List
      </a>
      <a
        class="nav--link"
        href="/form"
      >
        Add new event
      </a>
    </div>
    <div
      class="loading"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleDisableShrink MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
